helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterEventMonitor.java

```diff
@@ -57,7 +58,9 @@ public ClusterEventMonitor(ClusterStatusMonitor clusterStatusMonitor, String pha
     _clusterStatusMonitor = clusterStatusMonitor;
 
     _duration = new HistogramDynamicMetric("DurationGauge", new Histogram(
-        new SlidingTimeWindowArrayReservoir(DEFAULT_RESET_INTERVAL_MS, TimeUnit.MILLISECONDS)));
+        new SlidingTimeWindowArrayReservoir(HelixUtil
+            .getSystemPropertyAsLong(RESET_INTERVAL_SYSTEM_PROPERTY_KEY, DEFAULT_RESET_INTERVAL_MS),
```

Comment:
To avoid duplicate code, can you create a protected method in the parent class DynamicMBeanProvider to get the interval?
The logic looks fine to me.

Note the RESET_INTERVAL_SYSTEM_PROPERTY_KEY and DEFAULT_RESET_INTERVAL_MS are already there.